sale_stock_warehouse_partner/models/sale_order.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2025 Tecnativa - Eduardo Ezerouali
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    warehouse_id = fields.Many2one(
+        "stock.warehouse",
+        string="Warehouse",
+        required=True,
+        readonly=True,
+        states={"draft": [("readonly", False)], "sent": [("readonly", False)]},
+        compute="_compute_warehouse_id",
+        store=True,
+        check_company=True,
+    )
+
+    @api.depends("partner_shipping_id", "user_id")
+    def _compute_warehouse_id(self):
+        for order in self:
+            if order.partner_shipping_id.warehouse_id:
+                order.warehouse_id = order.partner_shipping_id.warehouse_id
+            elif order.partner_id.warehouse_id:
+                order.warehouse_id = order.partner_id.warehouse_id
+            else:
+                order.warehouse_id = order.user_id._get_default_warehouse_id()
+
+    def check_sale_order_type(self):
+        sale_order_type_module = self.env["ir.module.module"].search(
+            [("name", "=", "sale_order_type")]
+        )
+        if sale_order_type_module or sale_order_type_module.state == "installed":
```

Comment:
Use registry instead of search to minimize DB calls.
```suggestion
        if "sale_order_type" in self.env.registry._init_modules and "sale_stock_warehouse_order_type" not in self.env.registry._init_modules:
```

Reply:
No longer that need it, but good to know.